test/app/components/dry_validation_component.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require "dry-validation"
+
+class ValidationContract < Dry::Validation::Contract
+  params do
+    required(:required).filled(:string)
+  end
+end
+
+module DryValidations
+  def validate!
+    result = self.class.contract.call(component_params)
+
+    unless result.errors.empty?
+      full_messages = result.errors.to_h.map { |f, messages| "#{f}: #{messages.join(', ')}" }
+      raise RuntimeError.new(full_messages.join(", "))
+    end
+  end
+end
```

Comment:
I understand the 'real world' example argument but we're trying to test/prove that you can provide an alternative `validate!` method. This feels like a few more moving parts that we need to do that. Can we simplify (maybe not have to need to use dry-validation and force the reader to understand that gem to grok this test). 

I'm all for documenting using dry-validation in the Readme as a real example.